api/src/main/java/org/apache/iceberg/ScanTaskGroup.java

```diff
@@ -26,6 +26,24 @@
  * @param <T> the type of scan tasks
  */
 public interface ScanTaskGroup<T extends ScanTask> extends ScanTask {
+  /**
+   * Returns a grouping key for this task group.
+   *
+   * <p>Tasks may produce records that belong to a specific group of data (e.g. partition). It may
+   * be beneficial to preserve that grouping while combining tasks into groups if query engines can
+   * leverage that information to avoid shuffles. Grouping keys indicate how data is split between
```

Comment:
Nit: 
`It may be beneficial to preserve that grouping while combining tasks into groups if query engines can leverage that information to avoid shuffles.` ->
`Query engines could leverage the grouping key to avoid shuffles by combining tasks within the same group.` ?

Reply:
Reworked the doc.